tests/Faker.Tests/Common/NameTests.cs

```diff
@@ -25,5 +25,19 @@ public void Should_Get_Suffix()
 
 			Assert.That(new[] { suffix }, Is.SubsetOf(possibleSuffixes));
 		}
+
+		[Test]
+		[Repeat(100000)]
+		public void Compute_Time_Elapsed_By_The_First_Method()
+		{
+			Name.First();
```

Comment:
Resources [seems to be cached](http://stackoverflow.com/questions/15521204/performace-overheads-when-using-resource-files-resx). Here the problem is represented by the Split() method which has O(n) complexity, being n equal to the entries to be splitted. n is about 3.000 in the case of first name.
By using caches, complexity reduces to O(log n)O(1), where n is the number of cached cultures stored in the dictionary. This gives the huge performance gain we see.